### PR TITLE
Make `address` and `email` fields optional

### LINKED ISSUE #27: Make `address` optional when adding contacts
Under the [given implementation](https://se-education.org/addressbook-level3/UserGuide.html), contacts must be added with an `address`. Make this field optional.

### PR DESCRIPTION
**Pull request by chengjunyuan**
Fix #27 .

Made the `address` and `email` fields for adding contacts optional. Partial fix for 28 also, this is the first step towards removing the `email` field. Will remove `email` field in a separate PR.

This was implemented by adding an if-else statement in the `AddCommandParser` class.

### CODE REVIEW

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 1]
```diff
@@ -33,16 +33,28 @@ public AddCommand parse(String args) throws ParseException {
...
+            email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        } else {
+            email = ParserUtil.parseEmail("filler@filler.com");
+        }
+
+        Address address = null;
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            address = ParserUtil.parseAddress(argMultimap.getValue(PREFIX_ADDRESS).get());
+        } else {
+            address = ParserUtil.parseAddress("Filler Address");
```
Comment (minreiseah) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:52]: Why is "Filler Address" here?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java [hunk 2]
```diff
@@ -33,16 +33,28 @@ public AddCommand parse(String args) throws ParseException {
...
                 || !argMultimap.getPreamble().isEmpty()) {
             throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, AddCommand.MESSAGE_USAGE));
         }
 
-        argMultimap.verifyNoDuplicatePrefixesFor(PREFIX_NAME, PREFIX_PHONE, PREFIX_EMAIL, PREFIX_ADDRESS);
+        Email email = null;
+        if (argMultimap.getValue(PREFIX_EMAIL).isPresent()) {
+            email = ParserUtil.parseEmail(argMultimap.getValue(PREFIX_EMAIL).get());
+        } else {
+            email = ParserUtil.parseEmail("filler@filler.com");
```
Comment (minreiseah) [inline src/main/java/seedu/address/logic/parser/AddCommandParser.java:45]: What is the purpose of "filler@filler.com"?

File: src/main/java/seedu/address/logic/parser/AddCommandParser.java (hunk 1 above)
Comment (chengjunyuan) [reply src/main/java/seedu/address/logic/parser/AddCommandParser.java:52]: Can take this as a reply to both comments; when working I realized that deleting the fields breaks something somewhere else in the system and I haven't identified it yet, my workaround to making these fields "optional" wrt user input is to still fill them in with something on the backend so even if the user doesn't enter an email on the front end, it still works.

**Comment by legionlegion** [comment]
Closed, see #65 instead

**Pull request closed without merging**